src/utils.ts

```diff
@@ -63,8 +73,8 @@ function injectionCSSCodePlugin(cssToInject: string, styleId?: string, injectCod
         load(id: string) {
             if (id == cssInjectedByJsId) {
                 const cssCode = JSON.stringify(cssToInject.trim());
-                const injectFunction = injectCode || defaultInjectCode;
-                return injectFunction(cssCode, { styleId });
+                const injectFunction = injectCode || injectStyleTag;
+                return `(${injectFunction})(${cssCode}, ${JSON.stringify({ styleId })})`;
```

Comment:
We can add another parameter to directly support the custom function a user needs. This way, we are sure that this plugin adds the minimum code needed to complete its work.

Reply:
To confirm my understanding of this comment, the desire would be that if a user wants to provide code (like I have done above), we can accept that as a parameter, but otherwise, stick with string building for the default case?

I believe we might be able to go one step further using some AST transforms if you're open to adding in `babel` ?

Reply:
I would like to keep the plugin without any dependencies since the goal is very simple. so in my opinion this part can be left as it was. at most we can add another parameter to the plugin where a function can be used